src/zerocoin_v3.cpp

```diff
@@ -750,8 +750,9 @@ void CSigmaState::Containers::RemoveMint(sigma::PublicCoin const & pubCoin) {
     mint_info_container::const_iterator iter = mintedPubCoins.find(pubCoin);
     if (iter != mintedPubCoins.end()) {
         mintMetaInfo[iter->second.coinGroupId][iter->second.denomination] -= 1;
+        CMintedCoinInfo tmpMintInfo(iter->second);
         mintedPubCoins.erase(iter);
-        CheckSurgeCondition(iter->second.coinGroupId, iter->second.denomination);
+        CheckSurgeCondition(tmpMintInfo.coinGroupId, tmpMintInfo.denomination);
```

Comment:
If removal of a mint creates surge condition it would mean some problem other than surge (e.g. incorrect sigma state). Probably it deserves an assert. The only thing that can lead to the surge is new spend